src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/IPAddressUtil.cs

```diff
@@ -34,7 +34,7 @@ public static bool IsMulticast(IPAddress address)
         public static unsafe IPAddress GetIPAddressFromNativeInfo(Interop.Sys.IpAddressInfo* addressInfo)
         {
             byte[] ipBytes = new byte[addressInfo->NumAddressBytes];
-            fixed (byte* ipArrayPtr = ipBytes)
+            fixed (byte* ipArrayPtr = &ipBytes[0])
```

Comment:
I looked through the native code, and it looked like `NumAddressBytes` would only ever be 4 or 16, plus the fact that it was subsequently calling the `IPAddress` ctor (which throws if not those sizes) gave me some confidence, but you're right, we don't know for sure at this point. Reverted.